atlp-rwanda/Phantom-Backend-Pirates
# Project structure setup

###What does this PR do?
Set up the project structure to help start building RESTful web APIs using Express framework with best structure practices

###Description of Task to be completed?

1. Set a project structure
2. Set up a welcome endpoint
3. Configure ESLint for code analysis
4. Set API documentation with Swagger

###How should this be manually tested?
After cloning the repo, cd into it and first install all the packages used by running npm install in the your terminal.

1. To run code analysis run `npm run lint` in your terminal. This should analyze both all js and jsx extension files.
2. To run the development server run `npm run dev` in your terminal and `localhost:3000` in the browser
3. To test swagger documentation run `localhost:3000/api-docs` in your browser
![swagger-1](https://user-images.githubusercontent.com/16195731/154656928-ba753542-9731-4827-a81d-036ab2149b98.png)
![swagger-2](https://user-images.githubusercontent.com/16195731/154656936-b4864e80-2807-498f-a444-e48f48008aec.png)
![StartProject](https://user-images.githubusercontent.com/16195731/154656939-62262fbc-68d9-403d-86cb-a94cc14c5748.png)


Screenshots (if appropriate)
